JavaScript/JavaScriptSDK/ajax/ajax.ts

```diff
@@ -1,528 +1,382 @@
-                    self.callbackDuration = dateTime.GetDuration(self.responseFinishedTime, self.callbackFinishedTime);
-
-                    var timeStamps = [self.responseStartedTime, self.responseFinishedTime, self.callbackFinishedTime];
-                    for (var i = timeStamps.length - 1; i >= 0; i--) {
-                        if (timeStamps[i] !== null) {
-                            self.endTime = timeStamps[i];
-                            self.ajaxTotalDuration = dateTime.GetDuration(self.requestSentTime, self.endTime);
-                            break;
-                        }
-                    }
-                };
+                return originalOpen.apply(this, arguments);
             };
+        }
 
-            var Util = {
-                getAbsoluteUrl: (function () {
-                    var a;
-
-                    return function (url) {
-                        if (!a) a = document.createElement('a');
-                        a.href = url;
-
-                        return a.href;
-                    };
-                })()
-            }
-
-            ///<summary>The function object that provides ajax monitoring on the page</summary>
-            function ajaxMonitoring(appInsights: Microsoft.ApplicationInsights.AppInsights) {
-                var appInsights = appInsights;
-
-                var initiated = false;
-
-                ///<summary>The main function that needs to be called in order to start Ajax Monitoring</summary>
-                this.Init = function () {
-                    if (supportMonitoring()) {
-                        interceptOpen();
-                        interceptSetRequestHeader();
-                        interceptSend();
-                        interceptAbort();
-                        initiated = true;
+        private instrumentSetRequestHeader() {
+            var originalSetRequestHeader = XMLHttpRequest.prototype.setRequestHeader;
+            var ajaxMonitorInstance = this;
+            XMLHttpRequest.prototype.setRequestHeader = function (name, value) {
+                try {
+                    if (ajaxMonitorInstance.isMonitoredInstance(this)) {                    
+                        // 2 is the length of ": " which is added to each header
+                        this.ajaxData.requestSize += stringUtils.GetLength(name) + stringUtils.GetLength(value) + 2;
                     }
-                };
-
-
-                ///<summary>Function that returns property name which will identify that monitoring for given instance of XmlHttpRequest is disabled</summary>
-                this.GetDisabledPropertyName = function () {
-                    return getDisabledPropertyName();
-                };
-
-
-                function getDisabledPropertyName() {
-                    return "Microsoft_ApplicationInsights_BypassAjaxInstrumentation";
+                } catch (e) {
```

Comment:
It's intentional - I don't want to handle/swallow exceptions from user's code. Also the code I do wrap with try/catch can potentially throw - however it's more a safety pattern than a real need to catch errors here.